src/cargo/core/compiler/job_queue.rs

```diff
@@ -838,9 +837,10 @@ impl<'cfg> DrainState<'cfg> {
             let mut sender = FinishOnDrop {
                 messages: &messages,
                 id,
-                result: Err(format_err!("worker panicked")),
+                ok: false,
             };
-            sender.result = job.run(&state);
+            let result = job.run(&state);
+            sender.ok = true;
```

Comment:
Instead of using a boolean here could the message always be sent through the destructor and `result` is just changed to an `Option`?

Reply:
Done